llpc/context/llpcCompiler.cpp

```diff
@@ -975,6 +992,42 @@ bool Compiler::CanUseRelocatableGraphicsShaderElf(
     return useRelocatableShaderElf;
 }
 
+// =====================================================================================================================
+// Returns true if a compute pipeline can be built out of the given shader info.
+bool Compiler::CanUseRelocatableComputeShaderElf(
+    const PipelineShaderInfo* pShaderInfo   // Shader info for the pipeline to be built
+    ) const
+{
+    if (!llvm::cl::UseRelocatableShaderElf)
+    {
+        return false;
+    }
+
+    bool useRelocatableShaderElf = true;
+    if (useRelocatableShaderElf && pShaderInfo != nullptr)
+    {
+        const ShaderModuleData* pModuleData = reinterpret_cast<const ShaderModuleData*>(pShaderInfo->pModuleData);
+        if ((pModuleData != nullptr) && (pModuleData->binType != BinaryType::Spirv))
+        {
+            useRelocatableShaderElf = false;
+        }
+    }
+
+    if (useRelocatableShaderElf && cl::RelocatableShaderElfLimit != -1)
+    {
+        static uint32_t relocatableElfCounter = 0;
+        if (relocatableElfCounter >= cl::RelocatableShaderElfLimit)
+        {
+            useRelocatableShaderElf = false;
+        }
+        else
+        {
+            ++relocatableElfCounter;
+        }
+    }
+    return useRelocatableShaderElf;
+}
```

Comment:
This isn't really a complaint to you specifically, but this is the kind of function that really ought to be much shorter:
```
bool Compiler::CanUseRelocatableComputeShaderElf(
    const PipelineShaderInfo* pShaderInfo   // Shader info for the pipeline to be built
    ) const
{
    if (!llvm::cl::UseRelocatableShaderElf)
        return false;

    if (pShaderInfo) {
        const ShaderModuleData* pModuleData = reinterpret_cast<const ShaderModuleData*>(pShaderInfo->pModuleData);
        if (pModuleData && pModuleData->binType != BinaryType::Spirv)
            return false;
    }

    if (cl::RelocatableShaderElfLimit != -1) {
        static uint32_t relocatableElfCounter = 0;
        if (relocatableElfCounter >= cl::RelocatableShaderElfLimit)
            return false;
        ++relocatableElfCounter;
    }
    return true;
}
```
I can live with the code style being bad for the time being, but please remove the `useRelocatableShaderElf` and use early-returns instead, as that makes the code much clearer to read.